pymc/distributions/distribution.py

```diff
@@ -83,6 +85,64 @@
 PLATFORM = sys.platform
 
 
+def filter_RNGs(params):
+    return [p for p in params if not isinstance(p.type, (RandomType, RandomGeneratorType))]
```

Comment:
Since this ended up not being needed let's just inline it in the custom moment function